account_invoice_start_end_dates_move/models/account_move.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2023 Sergio Corato
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    start_date = fields.Date(index=True)
+    end_date = fields.Date(index=True)
+
+    @api.onchange("start_date", "end_date")
+    def _onchange_dates(self):
+        for line in self.line_ids.filtered(
+            lambda x: self.company_id.apply_dates_all_lines
+            or x.product_id.must_have_dates
+        ):
+            line.start_date = line.move_id.start_date
+            line.end_date = line.move_id.end_date
+
+    def write(self, vals):
+        res = super().write(vals)
+        if "start_date" in vals:
+            lines = self.mapped("line_ids").filtered(
```

Comment:
styling: `mapped` is useless here !

```suggestion
            lines = self.line_ids.filtered(
```

Reply:
I used `mapped` because `self` is possibly a recordset, otherwise I should use a `for`.

Reply:
it shouldn't be a matter, ie:


```python
>>> env["account.move"].search([]).line_ids.move_id.mapped("name")
['INV/2023/07/0004', 'INV/2023/07/0003', 'INV/2023/07/0002', 'INV/2023/07/0001', 'BILL/2023/07/0001', 'BNK1/2023/01/0007', 'BNK1/2023/01/0006', 'BNK1/2023/01/0005', 'BNK1/2023/01/0004', 'BNK1/2023/01/0003', 'BNK1/2023/01/0002', 'BNK1/2023/01/0001', 'BILL/2018/09/0001']
```